gradle-baseline-java/src/main/groovy/com/palantir/baseline/tasks/CheckJUnitDependencies.java

```diff
@@ -109,16 +109,7 @@ private void validateSourceSet(SourceSet ss, Test task) {
                                 + "'org.junit.jupiter:junit-jupiter' dependency "
                                 + "because tests use JUnit4 and useJUnitPlatform() is not enabled.");
             }
-        } else {
-            String compileClasspath = ss.getCompileClasspathConfigurationName();
-            boolean compilingAgainstOldJunit = hasDep(compileClasspath, CheckJUnitDependencies::isJunit4);
-            Preconditions.checkState(
```

Comment:
Rather than deleting this entirely, could we just emit it as a `warn` or `info` log line?

Reply:
I agree you don't want devs to write junit4 tests, but I don't think thats solved by enforcing that junit and junit-jupiter don't appear on the same classpath without the vintage engine. The earlier condition to check that there are no junit4 tests without the vintage engine is sufficient

Reply:
For example, dropwizard-testing contains both junit4 rules and junit5 extensions in the same JAR

Reply:
This is going for a slightly different goal though, this is trying to say once you think you've fully migrated to JUnit5, there should be a nice ratchet so that new devs don't start adding junit4 stuff back agian

Reply:
right, conceivably once you've fully migrated to junit5 you would no longer have the vintage engine and so if a user added a junit4 test the check would fail with `Tests may be silently not running! Some tests still use JUnit4, but Gradle has...`

Reply:
I was intending this to be the thing that encourages people to delete their `vintage engine` dependency though, otherwise I think these will litter our projects forever!

Reply:
I've updated the PR to log the message at info